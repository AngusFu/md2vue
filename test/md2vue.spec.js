const { join } = require('path')
const { readFileSync } = require('fs')
const source = readFileSync(join(__dirname, './common.md')).toString()
const md2vue = require('../')

let id = 1333
let title = 'foo'

const customMarkups = () => {
  const uid = `vue-demo-${id++}`
  return `<input id="${uid}" type="checkbox" /><label for="${uid}"></label>`
}

const documentInfo = {
  // eslint-disable-next-line
  head: new Function(`return { title: '${title}' }`),
  layout: 'component',
  directives: {
    'effect-only': {
      inserted (el) {
        console.log(el)
      }
    }
  }
}

it('transform markdown file to vue file correctly', async () => {
  const vue = await md2vue(source, {
    target: 'vue',
    highlight: 'prism',
    customMarkups,
    documentInfo
  })
  expect(vue).toMatchSnapshot()
})

it('transform markdown file to javascript correctly', async () => {
  const js = await md2vue(source, {
    target: 'js',
    componentName: 'common-comp',
    highlight: 'prism',
    customMarkups,
    documentInfo
  })
  expect(js).toMatchSnapshot()
})
