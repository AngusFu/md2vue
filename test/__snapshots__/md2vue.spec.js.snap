// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform markdown file to javascript correctly 1`] = `
"/* eslint-disable */
var moduleExports = (function (module) {
  'use strict';
  ;(function(){
  /* eslint-disable */
  const md2vuedemo0 = (function (module) {
  ;(function(){
    module.exports = {
      data: function data() {
        return {
          count: 0
        }
      },
      methods: {
        incr: function incr(delta) {
          this.count += delta
        }
      }
    }
  })()
  if (module.exports.__esModule) module.exports = module.exports.default
  var __vue__options__ = (typeof module.exports === \\"function\\"? module.exports.options: module.exports)
  if (__vue__options__.functional) {console.error(\\"[vueify] functional components are not supported and should be defined in plain js files using render functions.\\")}
  __vue__options__.render = function render () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('div',{staticClass:\\"vue-demo\\"},[_c('div',{staticClass:\\"wrapper\\"},[_c('button',{on:{\\"click\\":function($event){_vm.incr(-1)}}},[_vm._v(\\"-\\")]),_vm._v(\\" \\"),_c('input',{attrs:{\\"type\\":\\"text\\",\\"readonly\\":\\"\\"},domProps:{\\"value\\":_vm.count}}),_vm._v(\\" \\"),_c('button',{on:{\\"click\\":function($event){_vm.incr(+1)}}},[_vm._v(\\"+\\")])])])}
  __vue__options__.staticRenderFns = []
  __vue__options__._scopeId = \\"data-v-029571ac\\"
  if (module.hot) {(function () {  var hotAPI = require(\\"vue-hot-reload-api\\")
    hotAPI.install(require(\\"vue\\"), true)
    if (!hotAPI.compatible) return
    module.hot.accept()
    if (!module.hot.data) {
      hotAPI.createRecord(\\"data-v-029571ac\\", __vue__options__)
    } else {
      hotAPI.rerender(\\"data-v-029571ac\\", __vue__options__)
    }
  })()}
    return module.exports;
  })({});

  const md2vuedemo1 = (function (module) {
  ;(function(){
    module.exports = {
      methods: {
        click: function click() {
          alert('clicked!')
        }
      }
    }
  })()
  if (module.exports.__esModule) module.exports = module.exports.default
  var __vue__options__ = (typeof module.exports === \\"function\\"? module.exports.options: module.exports)
  if (__vue__options__.functional) {console.error(\\"[vueify] functional components are not supported and should be defined in plain js files using render functions.\\")}
  __vue__options__.render = function render () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('div',{staticClass:\\"vue-demo\\"},[_c('button',{on:{\\"click\\":_vm.click}},[_vm._v(\\"click\\")])])}
  __vue__options__.staticRenderFns = []
  __vue__options__._scopeId = \\"data-v-029571ad\\"
  if (module.hot) {(function () {  var hotAPI = require(\\"vue-hot-reload-api\\")
    hotAPI.install(require(\\"vue\\"), true)
    if (!hotAPI.compatible) return
    module.hot.accept()
    if (!module.hot.data) {
      hotAPI.createRecord(\\"data-v-029571ad\\", __vue__options__)
    } else {
      hotAPI.rerender(\\"data-v-029571ad\\", __vue__options__)
    }
  })()}
    return module.exports;
  })({});


  const __exports = {\\"head\\": function anonymous() {;return { title: 'foo' };},\\"layout\\": \\"component\\",\\"directives\\": {\\"effect-only\\": {\\"inserted\\": function inserted(el) {;console.log(el);;}}}};
  __exports.components = {
    'md2vuedemo0': md2vuedemo0,
    'md2vuedemo1': md2vuedemo1
  }
  module.exports = __exports;

  })()
  if (module.exports.__esModule) module.exports = module.exports.default
  var __vue__options__ = (typeof module.exports === \\"function\\"? module.exports.options: module.exports)
  if (__vue__options__.functional) {console.error(\\"[vueify] functional components are not supported and should be defined in plain js files using render functions.\\")}
  __vue__options__.render = function render () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('article',{staticClass:\\"markdown-body\\"},[_c('h2',[_vm._v(\\"Introduction to md2vue\\")]),_vm._m(0),_vm._v(\\" \\"),_c('blockquote',[_c('p',[_vm._v(\\"Markdown is a lightweight markup language with plain text formatting syntax. It is designed so that it can be converted to HTML and many other formats using a tool by the same name. —— \\"),_c('a',{attrs:{\\"href\\":\\"https://en.wikipedia.org/wiki/Markdown\\",\\"target\\":\\"_blank\\"}},[_vm._v(\\"wikipedia\\")])])]),_vm._v(\\" \\"),_c('h3',[_vm._v(\\"Inspiration\\")]),_vm._m(1),_vm._v(\\" \\"),_c('h3',[_vm._v(\\"Dev Dependencies\\")]),_vm._m(2),_vm._v(\\" \\"),_c('h3',[_vm._v(\\"Typography & UI\\")]),_vm._m(3),_vm._v(\\" \\"),_c('h3',[_vm._v(\\"Demo\\")]),_c('p',[_vm._v(\\"The following code in vue can be rendered into an real tiny vue app:\\")]),_vm._v(\\" \\"),_c('div',{staticClass:\\"vue-demo-block vue-demo-block-demo-only\\"},[_c('md2vuedemo0')],1),_vm._v(\\" \\"),_vm._m(4),_vm._v(\\" \\"),_c('div',{staticClass:\\"vue-demo-block\\"},[_c('md2vuedemo1'),_vm._v(\\" \\"),_c('input',{attrs:{\\"id\\":\\"vue-demo-1336\\",\\"type\\":\\"checkbox\\"}}),_c('label',{attrs:{\\"for\\":\\"vue-demo-1336\\"}}),_vm._v(\\" \\"),_vm._m(5)],1)],1)}
  __vue__options__.staticRenderFns = [function render () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('p',[_vm._v(\\"You can get raw text of this document \\"),_c('strong',[_c('a',{attrs:{\\"href\\":\\"/AngusFu/md2vue/blob/master/test/common.md\\"}},[_vm._v(\\"here\\")])]),_vm._v(\\".\\")])},function render () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('p',[_vm._v(\\"Inspired by \\"),_c('a',{attrs:{\\"href\\":\\"https://nuxtjs.org\\"}},[_vm._v(\\"nuxt\\")]),_vm._v(\\", \\"),_c('a',{attrs:{\\"href\\":\\"https://github.com/AngusFu/md2vue\\",\\"target\\":\\"_blank\\"}},[_vm._v(\\"md2vue\\")]),_vm._v(\\" was initially aimed at transforming markdown texts in our Vue project, which were used both for documentation and demostration.\\")])},function render () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('ul',[_c('li',[_c('a',{attrs:{\\"href\\":\\"/chjj/marked\\"}},[_vm._v(\\"marked\\")]),_vm._v(\\": A markdown parser and compiler. Built for speed.\\")]),_vm._v(\\" \\"),_c('li',[_c('a',{attrs:{\\"href\\":\\"https://github.com/isagalaev/highlight.js\\",\\"target\\":\\"_blank\\"}},[_vm._v(\\"highlight.js\\")]),_vm._v(\\": Javascript syntax highlighter.\\")]),_vm._v(\\" \\"),_c('li',[_c('a',{attrs:{\\"href\\":\\"https://github.com/vuejs/vueify\\",\\"target\\":\\"_blank\\"}},[_vm._v(\\"vueify\\")]),_vm._v(\\": Browserify transform for single-file Vue components\\")]),_vm._v(\\" \\"),_c('li',[_c('a',{attrs:{\\"href\\":\\"https://github.com/rollup/rollup\\",\\"target\\":\\"_blank\\"}},[_vm._v(\\"rollup\\")]),_vm._v(\\": Next-generation ES6 module bundler.\\")]),_vm._v(\\" \\"),_c('li',[_c('a',{attrs:{\\"href\\":\\"https://buble.surge.sh/\\",\\"target\\":\\"_blank\\"}},[_vm._v(\\"buble\\")]),_vm._v(\\": The blazing fast, batteries-included ES2015 compiler.\\")])])},function render () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('ul',[_c('li',[_c('a',{attrs:{\\"href\\":\\"https://github.com/sindresorhus/github-markdown-css\\",\\"target\\":\\"_blank\\"}},[_vm._v(\\"github-markdown-css\\")]),_vm._v(\\": The minimal amount of CSS to replicate the GitHub Markdown style.\\")]),_vm._v(\\" \\"),_c('li',[_c('a',{attrs:{\\"href\\":\\"https://github.com/isagalaev/highlight.js\\",\\"target\\":\\"_blank\\"}},[_vm._v(\\"highlight.js\\")]),_vm._v(\\": Code highlighting.\\")])])},function render () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('p',[_vm._v(\\"You can also leave out \\"),_c('code',[_vm._v(\\"<template>\\")]),_vm._v(\\" tags, just like this:\\")])},function render () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('pre',{pre:true,attrs:{\\"class\\":\\"lang-html\\"}},[_c('code',[_c('span',{attrs:{\\"class\\":\\"token tag\\"}},[_c('span',{attrs:{\\"class\\":\\"token tag\\"}},[_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\"<\\")]),_vm._v(\\"style\\")]),_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\">\\")])]),_c('span',{attrs:{\\"class\\":\\"token style language-css\\"}},[_vm._v(\\"\\\\n  \\"),_c('span',{attrs:{\\"class\\":\\"token selector\\"}},[_vm._v(\\"button\\")]),_vm._v(\\" \\"),_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\"{\\")]),_vm._v(\\"\\\\n    \\"),_c('span',{attrs:{\\"class\\":\\"token property\\"}},[_vm._v(\\"font-size\\")]),_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\":\\")]),_vm._v(\\" 14px\\"),_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\";\\")]),_vm._v(\\"\\\\n  \\"),_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\"}\\")]),_vm._v(\\"\\\\n\\")]),_c('span',{attrs:{\\"class\\":\\"token tag\\"}},[_c('span',{attrs:{\\"class\\":\\"token tag\\"}},[_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\"</\\")]),_vm._v(\\"style\\")]),_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\">\\")])]),_vm._v(\\"\\\\n\\"),_c('span',{attrs:{\\"class\\":\\"token tag\\"}},[_c('span',{attrs:{\\"class\\":\\"token tag\\"}},[_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\"<\\")]),_vm._v(\\"button\\")]),_vm._v(\\" \\"),_c('span',{attrs:{\\"class\\":\\"token attr-name\\"}},[_vm._v(\\"@click\\")]),_c('span',{attrs:{\\"class\\":\\"token attr-value\\"}},[_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\"=\\")]),_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\"\\\\\\"\\")]),_vm._v(\\"click\\"),_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\"\\\\\\"\\")])]),_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\">\\")])]),_vm._v(\\"click\\"),_c('span',{attrs:{\\"class\\":\\"token tag\\"}},[_c('span',{attrs:{\\"class\\":\\"token tag\\"}},[_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\"</\\")]),_vm._v(\\"button\\")]),_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\">\\")])]),_vm._v(\\"\\\\n\\\\n\\"),_c('span',{attrs:{\\"class\\":\\"token tag\\"}},[_c('span',{attrs:{\\"class\\":\\"token tag\\"}},[_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\"<\\")]),_vm._v(\\"script\\")]),_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\">\\")])]),_c('span',{attrs:{\\"class\\":\\"token script language-javascript\\"}},[_vm._v(\\"\\\\n  \\"),_c('span',{attrs:{\\"class\\":\\"token keyword\\"}},[_vm._v(\\"export\\")]),_vm._v(\\" \\"),_c('span',{attrs:{\\"class\\":\\"token keyword\\"}},[_vm._v(\\"default\\")]),_vm._v(\\" \\"),_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\"{\\")]),_vm._v(\\"\\\\n    methods\\"),_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\":\\")]),_vm._v(\\" \\"),_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\"{\\")]),_vm._v(\\"\\\\n      \\"),_c('span',{attrs:{\\"class\\":\\"token function\\"}},[_vm._v(\\"click\\")]),_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\"(\\")]),_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\")\\")]),_vm._v(\\" \\"),_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\"{\\")]),_vm._v(\\"\\\\n        \\"),_c('span',{attrs:{\\"class\\":\\"token function\\"}},[_vm._v(\\"alert\\")]),_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\"(\\")]),_c('span',{attrs:{\\"class\\":\\"token string\\"}},[_vm._v(\\"'clicked!'\\")]),_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\")\\")]),_vm._v(\\"\\\\n      \\"),_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\"}\\")]),_vm._v(\\"\\\\n    \\"),_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\"}\\")]),_vm._v(\\"\\\\n  \\"),_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\"}\\")]),_vm._v(\\"\\\\n\\")]),_c('span',{attrs:{\\"class\\":\\"token tag\\"}},[_c('span',{attrs:{\\"class\\":\\"token tag\\"}},[_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\"</\\")]),_vm._v(\\"script\\")]),_c('span',{attrs:{\\"class\\":\\"token punctuation\\"}},[_vm._v(\\">\\")])])])])}]
  if (module.hot) {(function () {  var hotAPI = require(\\"vue-hot-reload-api\\")
    hotAPI.install(require(\\"vue\\"), true)
    if (!hotAPI.compatible) return
    module.hot.accept()
    if (!module.hot.data) {
      hotAPI.createRecord(\\"data-v-04979ffb\\", __vue__options__)
    } else {
      hotAPI.reload(\\"data-v-04979ffb\\", __vue__options__)
    }
  })()}
  var exports = module.exports
  exports.name = \\"common-comp\\"
  exports.created = function () {
    const css = \\".wrapper input[data-v-029571ac] {     width: 50px;     text-align: center;   } button[data-v-029571ad] {     font-size: 14px;   }\\"
    if (css) {
      this.____ = insert(css)
    }
  }
  exports.destroyed = function () {
    this.____ && this.____()
  }
  module.exports.install = function (Vue) {
    Vue.component(exports.name, exports)
  }

  return module.exports;

  function insert(css) {
    var elem = document.createElement('style')
    elem.setAttribute('type', 'text/css')

    if ('textContent' in elem) {
      elem.textContent = css
    } else {
      elem.styleSheet.cssText = css
    }

    document.getElementsByTagName('head')[0].appendChild(elem)
    return function () {
      document.getElementsByTagName('head')[0].removeChild(elem)
    }
  }
})({});
typeof exports === 'object' && typeof module !== 'undefined' && (module.exports = moduleExports);
typeof window !== void 0 && window.Vue && Vue.use(moduleExports);
this[\\"CommonComp\\"] = moduleExports;
"
`;

exports[`transform markdown file to vue file correctly 1`] = `
"
<style>
.wrapper input[data-v-029571ac] {
    width: 50px;
    text-align: center;
  }
button[data-v-029571ad] {
    font-size: 14px;
  }
</style>

<template>
<article class=\\"markdown-body\\">
<h2>Introduction to md2vue</h2><p>You can get raw text of this document <strong><a href=\\"/AngusFu/md2vue/blob/master/test/common.md\\">here</a></strong>.</p>
<blockquote>
<p>Markdown is a lightweight markup language with plain text formatting syntax. It is designed so that it can be converted to HTML and many other formats using a tool by the same name. —— <a href=\\"https://en.wikipedia.org/wiki/Markdown\\" target=\\"_blank\\">wikipedia</a></p>
</blockquote>
<h3>Inspiration</h3><p>Inspired by <a href=\\"https://nuxtjs.org\\">nuxt</a>, <a href=\\"https://github.com/AngusFu/md2vue\\" target=\\"_blank\\">md2vue</a> was initially aimed at transforming markdown texts in our Vue project, which were used both for documentation and demostration.</p>
<h3>Dev Dependencies</h3><ul>
<li><a href=\\"/chjj/marked\\">marked</a>: A markdown parser and compiler. Built for speed.</li>
<li><a href=\\"https://github.com/isagalaev/highlight.js\\" target=\\"_blank\\">highlight.js</a>: Javascript syntax highlighter.</li>
<li><a href=\\"https://github.com/vuejs/vueify\\" target=\\"_blank\\">vueify</a>: Browserify transform for single-file Vue components</li>
<li><a href=\\"https://github.com/rollup/rollup\\" target=\\"_blank\\">rollup</a>: Next-generation ES6 module bundler.</li>
<li><a href=\\"https://buble.surge.sh/\\" target=\\"_blank\\">buble</a>: The blazing fast, batteries-included ES2015 compiler.</li>
</ul>
<h3>Typography &amp; UI</h3><ul>
<li><a href=\\"https://github.com/sindresorhus/github-markdown-css\\" target=\\"_blank\\">github-markdown-css</a>: The minimal amount of CSS to replicate the GitHub Markdown style.</li>
<li><a href=\\"https://github.com/isagalaev/highlight.js\\" target=\\"_blank\\">highlight.js</a>: Code highlighting.</li>
</ul>
<h3>Demo</h3><p>The following code in vue can be rendered into an real tiny vue app:</p>

<div class=\\"vue-demo-block vue-demo-block-demo-only\\">
<md2vuedemo0></md2vuedemo0>


</div>
<p>You can also leave out <code>&lt;template&gt;</code> tags, just like this:</p>

<div class=\\"vue-demo-block\\">
<md2vuedemo1></md2vuedemo1>
<input id=\\"vue-demo-1334\\" type=\\"checkbox\\" /><label for=\\"vue-demo-1334\\"></label>
<pre v-pre class=\\"lang-html\\"><code><span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&lt;</span>style</span><span class=\\"token punctuation\\">></span></span><span class=\\"token style language-css\\">
  <span class=\\"token selector\\">button</span> <span class=\\"token punctuation\\">{</span>
    <span class=\\"token property\\">font-size</span><span class=\\"token punctuation\\">:</span> 14px<span class=\\"token punctuation\\">;</span>
  <span class=\\"token punctuation\\">}</span>
</span><span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&lt;/</span>style</span><span class=\\"token punctuation\\">></span></span>
<span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&lt;</span>button</span> <span class=\\"token attr-name\\">@click</span><span class=\\"token attr-value\\"><span class=\\"token punctuation\\">=</span><span class=\\"token punctuation\\">\\"</span>click<span class=\\"token punctuation\\">\\"</span></span><span class=\\"token punctuation\\">></span></span>click<span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&lt;/</span>button</span><span class=\\"token punctuation\\">></span></span>

<span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&lt;</span>script</span><span class=\\"token punctuation\\">></span></span><span class=\\"token script language-javascript\\">
  <span class=\\"token keyword\\">export</span> <span class=\\"token keyword\\">default</span> <span class=\\"token punctuation\\">{</span>
    methods<span class=\\"token punctuation\\">:</span> <span class=\\"token punctuation\\">{</span>
      <span class=\\"token function\\">click</span><span class=\\"token punctuation\\">(</span><span class=\\"token punctuation\\">)</span> <span class=\\"token punctuation\\">{</span>
        <span class=\\"token function\\">alert</span><span class=\\"token punctuation\\">(</span><span class=\\"token string\\">'clicked!'</span><span class=\\"token punctuation\\">)</span>
      <span class=\\"token punctuation\\">}</span>
    <span class=\\"token punctuation\\">}</span>
  <span class=\\"token punctuation\\">}</span>
</span><span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&lt;/</span>script</span><span class=\\"token punctuation\\">></span></span></code></pre>
</div>

</article >
</template>

<script>
/* eslint-disable */
const md2vuedemo0 = (function (module) {
;(function(){
  module.exports = {
    data: function data() {
      return {
        count: 0
      }
    },
    methods: {
      incr: function incr(delta) {
        this.count += delta
      }
    }
  }
})()
if (module.exports.__esModule) module.exports = module.exports.default
var __vue__options__ = (typeof module.exports === \\"function\\"? module.exports.options: module.exports)
if (__vue__options__.functional) {console.error(\\"[vueify] functional components are not supported and should be defined in plain js files using render functions.\\")}
__vue__options__.render = function render () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('div',{staticClass:\\"vue-demo\\"},[_c('div',{staticClass:\\"wrapper\\"},[_c('button',{on:{\\"click\\":function($event){_vm.incr(-1)}}},[_vm._v(\\"-\\")]),_vm._v(\\" \\"),_c('input',{attrs:{\\"type\\":\\"text\\",\\"readonly\\":\\"\\"},domProps:{\\"value\\":_vm.count}}),_vm._v(\\" \\"),_c('button',{on:{\\"click\\":function($event){_vm.incr(+1)}}},[_vm._v(\\"+\\")])])])}
__vue__options__.staticRenderFns = []
__vue__options__._scopeId = \\"data-v-029571ac\\"
if (module.hot) {(function () {  var hotAPI = require(\\"vue-hot-reload-api\\")
  hotAPI.install(require(\\"vue\\"), true)
  if (!hotAPI.compatible) return
  module.hot.accept()
  if (!module.hot.data) {
    hotAPI.createRecord(\\"data-v-029571ac\\", __vue__options__)
  } else {
    hotAPI.reload(\\"data-v-029571ac\\", __vue__options__)
  }
})()}
  return module.exports;
})({});

const md2vuedemo1 = (function (module) {
;(function(){
  module.exports = {
    methods: {
      click: function click() {
        alert('clicked!')
      }
    }
  }
})()
if (module.exports.__esModule) module.exports = module.exports.default
var __vue__options__ = (typeof module.exports === \\"function\\"? module.exports.options: module.exports)
if (__vue__options__.functional) {console.error(\\"[vueify] functional components are not supported and should be defined in plain js files using render functions.\\")}
__vue__options__.render = function render () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('div',{staticClass:\\"vue-demo\\"},[_c('button',{on:{\\"click\\":_vm.click}},[_vm._v(\\"click\\")])])}
__vue__options__.staticRenderFns = []
__vue__options__._scopeId = \\"data-v-029571ad\\"
if (module.hot) {(function () {  var hotAPI = require(\\"vue-hot-reload-api\\")
  hotAPI.install(require(\\"vue\\"), true)
  if (!hotAPI.compatible) return
  module.hot.accept()
  if (!module.hot.data) {
    hotAPI.createRecord(\\"data-v-029571ad\\", __vue__options__)
  } else {
    hotAPI.reload(\\"data-v-029571ad\\", __vue__options__)
  }
})()}
  return module.exports;
})({});


const __exports = {\\"head\\": function anonymous() {;return { title: 'foo' };},\\"layout\\": \\"component\\",\\"directives\\": {\\"effect-only\\": {\\"inserted\\": function inserted(el) {;console.log(el);;}}}};
__exports.components = {
  'md2vuedemo0': md2vuedemo0,
  'md2vuedemo1': md2vuedemo1
}
module.exports = __exports;
</script>"
`;
